### PR TITLE
CLUS-4438 serving ccmgr.js and enableMcc endpoint

### PR DESCRIPTION
**Pull request by pablopnn**
jiras:
https://severalnines.atlassian.net/browse/CLUS-4438
https://severalnines.atlassian.net/browse/CLUS-4701

- Serve `/ccmgr.js`
- Serve `/mcc/enable` endpoint
- Added ccmgradm command enableMcc
- Enable proxy authentication for `/v2` paths only when mcc is enabled
- first admin registration

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -63,33 +64,35 @@ type CmonInstance struct {
...
+	Instances       	[]*CmonInstance `yaml:"instances,omitempty"`
+	Timeout         	int             `yaml:"timeout,omitempty"`
+	Logfile         	string          `yaml:"logfile,omitempty"`
+	Users           	[]*ProxyUser    `yaml:"users,omitempty"`
+	FrontendPath    	string          `yaml:"frontend_path,omitempty" json:"frontend_path,omitempty"`
+	Port            	int             `yaml:"port" json:"port"`
+	TlsCert         	string          `yaml:"tls_cert,omitempty" json:"tls_cert,omitempty"`
+	TlsKey          	string          `yaml:"tls_key,omitempty" json:"tls_key,omitempty"`
+	SessionTtl      	int64           `yaml:"session_ttl" json:"session_ttl"` // in nanoseconds, min 30 minutes
+	SingleController 	string		    `yaml:"single_controller" json:"single_controller"` // in nanoseconds, min 30 minutes
```
Comment (alexgolod) [inline config/config.go:80]: could this be bool instead, or ideas was to store xid here in the feature?

Comment (alexgolod) [reply config/config.go:80]: oh, right, it is xid

File: rpcserver/server.go [hunk 2]
```diff
@@ -322,7 +354,9 @@ func Start(cfg *config.Config) {
 	// Proxy any /v2 requests to the specified (by controller_id) cmon
 	v2 := s.Group("/v2")
 	{
-		v2.Use(proxy.RPCAuthMiddleware)
+		if (cfg.SingleController == "") {
```
Comment (alexgolod) [inline rpcserver/server.go:357]: What was the problem with auth?

File: rpcserver/server.go [hunk 3]
```diff
@@ -142,6 +142,38 @@ func serveFrontend(s *gin.Engine, cfg *config.Config) error {
...
+				jsonStr := string(jsonBytes)
+
+				// Create the JavaScript response
+				jsContent := fmt.Sprintf("window.CCMGR = %s;", jsonStr)
+				c.Header("Content-Type", "application/javascript");
+				c.String(http.StatusOK, jsContent)
+				c.Abort()
+				return
+			}
+			if c.Request.URL.Path == "/enableMcc" {
```
Comment (alexgolod) [inline rpcserver/server.go:170]: I think this might fit better outside of serveFrontend

Comment (alexgolod) [reply rpcserver/server.go:170]: Maybe could be /proxy/enableMcc ?

File: config/config.go [hunk 4]
```diff
@@ -63,33 +64,35 @@ type CmonInstance struct {
...
+	Instances        []*CmonInstance `yaml:"instances,omitempty"`
+	Timeout          int             `yaml:"timeout,omitempty"`
+	Logfile          string          `yaml:"logfile,omitempty"`
+	Users            []*ProxyUser    `yaml:"users,omitempty"`
+	FrontendPath     string          `yaml:"frontend_path,omitempty" json:"frontend_path,omitempty"`
+	Port             int             `yaml:"port" json:"port"`
+	TlsCert          string          `yaml:"tls_cert,omitempty" json:"tls_cert,omitempty"`
+	TlsKey           string          `yaml:"tls_key,omitempty" json:"tls_key,omitempty"`
+	SessionTtl       int64           `yaml:"session_ttl" json:"session_ttl"`             // in nanoseconds, min 30 minutes
+	SingleController string          `yaml:"single_controller" json:"single_controller"` // in nanoseconds, min 30 minutes
```
Comment (bunin) [inline config/config.go:80]:
```suggestion
	SingleController string          `yaml:"single_controller" json:"single_controller"`
```

**Review by bunin — APPROVED** [review]

**Review by pablopnn — COMMENTED** [review]
Looks good

**Pull request merged**